### ISSUE TITLE
[BUG] v0.5.2 TabularPredictor Feature importance method fail on quantile regression problem type

### ISSUE THREAD

**Issue by astro313** [awslabs/autogluon#2261]
- [x] I have checked that this bug exists on the latest stable version of AutoGluon
- [ ] and/or I have checked that this bug exists on the latest mainline of AutoGluon via source installation

**Describe the bug**
I trained a TabularPredictor using quantile regression (`problem_type='quantile'`). Tried `predictor.feature_importance(data)`, but got an error:
```~/anaconda3/envs/pytorch_p38/lib/python3.8/site-packages/autogluon/tabular/predictor/predictor.py in feature_importance(self, data, model, features, feature_stage, subsample_size, time_limit, num_shuffle_sets, include_confidence_band, confidence_level, silent)
   1960             num_shuffle_sets = 10 if time_limit else 5
   1961 
-> 1962         fi_df = self._learner.get_feature_importance(model=model, X=data, features=features,
   1963                                                      feature_stage=feature_stage,
   1964                                                      subsample_size=subsample_size, time_limit=time_limit,

~/anaconda3/envs/pytorch_p38/lib/python3.8/site-packages/autogluon/tabular/learner/abstract_learner.py in get_feature_importance(self, model, X, y, features, feature_stage, subsample_size, silent, **kwargs)
    666 
    667             if feature_stage == 'original':
--> 668                 return trainer._get_feature_importance_raw(model=model, X=X, y=y, features=features, subsample_size=subsample_size, transform_func=self.transform_features, silent=silent, **kwargs)
    669             X = self.transform_features(X)
    670         else:

~/anaconda3/envs/pytorch_p38/lib/python3.8/site-packages/autogluon/core/trainer/abstract_trainer.py in _get_feature_importance_raw(self, X, y, model, eval_metric, **kwargs)
   1888         model: AbstractModel = self.load_model(model)
   1889         predict_func_kwargs = dict(model=model)
-> 1890         return compute_permutation_feature_importance(
   1891             X=X, y=y, predict_func=predict_func, predict_func_kwargs=predict_func_kwargs, eval_metric=eval_metric, **kwargs
   1892         )

~/anaconda3/envs/pytorch_p38/lib/python3.8/site-packages/autogluon/core/utils/utils.py in compute_permutation_feature_importance(X, y, predict_func, eval_metric, features, subsample_size, num_shuffle_sets, predict_func_kwargs, transform_func, transform_func_kwargs, time_limit, silent, log_prefix, importance_as_list, random_state)
    710             X_transformed = X if transform_func is None else transform_func(X, **transform_func_kwargs)
    711             y_pred = predict_func(X_transformed, **predict_func_kwargs)
--> 712             score_baseline = eval_metric(y, y_pred)
    713             if shuffle_repeat == 0:
    714                 if not silent:

~/anaconda3/envs/pytorch_p38/lib/python3.8/site-packages/autogluon/core/metrics/__init__.py in __call__(self, y_true, y_pred, sample_weight, **kwargs)
     83             k['sample_weight'] = sample_weight
     84 
---> 85         return self._score(y_true=y_true, y_pred=y_pred, **k)
     86 
     87     def _score(self, y_true, y_pred, **kwargs):

~/anaconda3/envs/pytorch_p38/lib/python3.8/site-packages/autogluon/core/metrics/__init__.py in _score(self, y_true, y_pred, **kwargs)
     86 
     87     def _score(self, y_true, y_pred, **kwargs):
---> 88         y_true, y_pred, kwargs = self._preprocess(y_true=y_true, y_pred=y_pred, **kwargs)
     89         return self._sign * self._score_func(y_true, y_pred, **kwargs)
     90 

~/anaconda3/envs/pytorch_p38/lib/python3.8/site-packages/autogluon/core/metrics/__init__.py in _preprocess(self, y_true, y_pred, **kwargs)
    246             y_pred = np.array(y_pred)
    247         if 'quantile_levels' not in kwargs:
--> 248             raise AssertionError('quantile_levels is required to score quantile metrics')
    249         if isinstance(kwargs['quantile_levels'], list):
    250             kwargs['quantile_levels'] = np.array(kwargs['quantile_levels'])

AssertionError: quantile_levels is required to score quantile metrics
``` 

When providing quantile_levels to this method, it returns
```
TypeError: feature_importance() got an unexpected keyword argument 'quantile_levels'
```

**Installed Versions**
0.5.2

**Comment by Innixma** [comment]
Thanks for reporting! This is indeed a bug. We can try to get it fixed for v0.6 release.

**Comment by Innixma** [comment]
This seems to be fixed in mainline / upcoming v0.6:

```python3
from autogluon.tabular import TabularDataset, TabularPredictor
from random import uniform

train_data = TabularDataset('https://autogluon.s3.amazonaws.com/datasets/Inc/train.csv')
label = 'class'

# remove the classification label
train_data.drop(columns=['class'])
#add the regression labels containing  0 - 100 as floats with 2 decimal places
train_data['class'] = [round(uniform(0, 100), 2) for i in range(len(train_data))]

predictor = TabularPredictor(label=label, problem_type='quantile', quantile_levels=[0.4, 0.5, 0.6]).fit(train_data, time_limit=10)
predictor.feature_importance(train_data)

```

```
No path specified. Models will be saved in: "AutogluonModels/ag-20221108_203152/"
Beginning AutoGluon training ... Time limit = 10s
AutoGluon will save models to "AutogluonModels/ag-20221108_203152/"
AutoGluon Version:  0.5.3b20221101
Python Version:     3.8.10
Operating System:   Darwin
Platform Machine:   x86_64
Platform Version:   Darwin Kernel Version 21.6.0: Thu Sep 29 20:12:57 PDT 2022; root:xnu-8020.240.7~1/RELEASE_X86_64
Train Data Rows:    39073
Train Data Columns: 14
Label Column: class
Preprocessing data ...
Using Feature Generators to preprocess the data ...
Fitting AutoMLPipelineFeatureGenerator...
	Available Memory:                    7283.42 MB
	Train Data (Original)  Memory Usage: 22.92 MB (0.3% of available memory)
	Inferring data type of each feature based on column values. Set feature_metadata_in to manually specify special dtypes of the features.
	Stage 1 Generators:
		Fitting AsTypeFeatureGenerator...
			Note: Converting 1 features to boolean dtype as they only contain 2 unique values.
	Stage 2 Generators:
		Fitting FillNaFeatureGenerator...
	Stage 3 Generators:
		Fitting IdentityFeatureGenerator...
		Fitting CategoryFeatureGenerator...
			Fitting CategoryMemoryMinimizeFeatureGenerator...
	Stage 4 Generators:
		Fitting DropUniqueFeatureGenerator...
	Types of features in original data (raw dtype, special dtypes):
		('int', [])    : 6 | ['age', 'fnlwgt', 'education-num', 'capital-gain', 'capital-loss', ...]
		('object', []) : 8 | ['workclass', 'education', 'marital-status', 'occupation', 'relationship', ...]
	Types of features in processed data (raw dtype, special dtypes):
		('category', [])  : 7 | ['workclass', 'education', 'marital-status', 'occupation', 'relationship', ...]
		('int', [])       : 6 | ['age', 'fnlwgt', 'education-num', 'capital-gain', 'capital-loss', ...]
		('int', ['bool']) : 1 | ['sex']
	0.2s = Fit runtime
	14 features in original data used to generate 14 features in processed data.
	Train Data (Processed) Memory Usage: 2.19 MB (0.0% of available memory)
Data preprocessing and feature engineering runtime = 0.25s ...
AutoGluon will gauge predictive performance using evaluation metric: 'pinball_loss'
	This metric's sign has been flipped to adhere to being higher_is_better. The metric score can be multiplied by -1 to get the metric value.
	To change this, specify the eval_metric parameter of Predictor()
Automatically generating train/validation split with holdout_frac=0.0639828014229775, Train Rows: 36573, Val Rows: 2500
Fitting 4 L1 models ...
Fitting model: RandomForestMSE ... Training model for up to 9.75s of the 9.75s of remaining time.
	WARNING: RandomForestQuantileRegressor are experimental for quantile regression. They may change or be removed without warning in future releases.
	-13.0175	 = Validation score   (-pinball_loss)
	11.78s	 = Training   runtime
	3.37s	 = Validation runtime
Fitting model: WeightedEnsemble_L2 ... Training model for up to 9.75s of the -6.3s of remaining time.
	-13.0175	 = Validation score   (-pinball_loss)
	0.0s	 = Training   runtime
	0.0s	 = Validation runtime
AutoGluon training complete, total runtime = 16.33s ... Best model: "WeightedEnsemble_L2"
TabularPredictor saved. To load, use: predictor = TabularPredictor.load("AutogluonModels/ag-20221108_203152/")
Computing feature importance via permutation shuffling for 14 features using 5000 rows with 5 shuffle sets...
	448.4s	= Expected runtime (89.68s per shuffle set)
	320.95s	= Actual runtime (Completed 5 of 5 shuffle sets)
```
